src/zenml/cli/stack_components.py

```diff
@@ -221,6 +245,123 @@ def register_stack_component_command(
+            StackComponentClassRegistry,
+        )
+
+        component_class = StackComponentClassRegistry.get_class(
+            component_type=component_type,
+            component_flavor=current_component.FLAVOR,
+        )
+        available_properties = _get_available_properties(component_class)
+        for prop in parsed_args.keys():
+            if (prop not in available_properties) and (
+                len(available_properties) > 0
+            ):
+                cli_utils.error(
+                    f"You cannot update the {display_name} `{current_component.name}` with property '{prop}'. You can only update the following properties: {available_properties}."
+                )
+            elif prop not in available_properties:
+                cli_utils.error(
+                    f"You cannot update the {display_name} `{current_component.name}` with property '{prop}' as this {display_name} has no optional properties that can be configured."
+                )
+            else:
+                continue
+
+        required_properties = _get_required_properties(component_class)
+        for prop in required_properties:
+            if prop not in parsed_args:
+                parsed_args[prop] = getattr(current_component, prop)
+        updated_component = component_class(name=name, **parsed_args)
+
+        repo.update_stack_component(updated_component)
+        cli_utils.declare(f"Successfully updated {display_name} `{name}`.")
+
+    return update_stack_component_command
+
+
+def generate_stack_component_rename_command(
+    component_type: StackComponentType,
+) -> Callable[[str, str], None]:
+    """Generates a `rename` command for the specific stack component type."""
+    display_name = _component_display_name(component_type)
+
+    @click.argument(
+        "name",
+        type=str,
+        required=True,
+    )
+    @click.argument(
+        "new_name",
+        type=str,
+        required=True,
+    )
+    def rename_stack_component_command(name: str, new_name: str) -> None:
+        """Rename a stack component."""
+        repo = Repository()
+        cli_utils.print_active_profile()
+        current_component = repo.get_stack_component(component_type, name)
+        if current_component is None:
+            cli_utils.error(f"No {display_name} found for name '{name}'.")
+
+        from zenml.stack.stack_component_class_registry import (
+            StackComponentClassRegistry,
+        )
+
+        component_class = StackComponentClassRegistry.get_class(
+            component_type=component_type,
+            component_flavor=current_component.FLAVOR,
+        )
+
+        required_properties = _get_required_properties(component_class)
+        extra_args = {}
+        for prop in required_properties:
+            if prop not in extra_args:
+                extra_args[prop] = getattr(current_component, prop)
+        renamed_component = component_class(name=new_name, **extra_args)
+
```

Comment:
This goes alongside what I was saying earlier about the `rename_stack_component` signature: why do you need to go to all this trouble of copying attributes from the existing component configuration over to the updated component configuration, if all you need is to change a single attribute: the name ? Surely, this could be done easier by just passing the `old_name` and `new_name` to `rename_stack_component` and let the store implementation deal with all this complexity.

Reply:
This is because there are fundamentally two different ways we handle the updating / renaming. For the local stack store we need the wrapper / etc, but the SQL/REST stack stores are much smarter about how they can access the current state of the components.

It's a bit ugly, but a lot of this is just me inheriting from the way we currently do things, particularly in the local stack store.